File: 06-03-18/c++/02-Micro-and-Array-Update.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        int n,k,mini=INT_MAX;
        cin>>n>>k;
        int ar[n];
        for(int i=0;i<n;i++)
        {
            cin>>ar[i];
            mini=min(ar[i],mini);
            
        }
        cout<<max(0,k-mini)<<endl;
        
    }
}

File: 06-03-18/java/02-Micro-and-Array-Update.java
import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        
        int t;
        t=input.nextInt();
        while(t-->0)
        {
            int n,k,mini=1000001;
            n=input.nextInt();
            k=input.nextInt();
            int ar[]=new int[n];
            for(int i=0;i<n;i++)
            {
                ar[i]=input.nextInt();
                mini=Math.min(ar[i],mini);
                    
            }
            System.out.println(Math.max(0,k-mini));
            
        }
    }
}
